flink-core/src/test/java/org/apache/flink/api/common/io/DelimitedInputFormatTest.java

```diff
@@ -431,6 +431,34 @@ public void testDelimiterOnBufferBoundary() throws IOException {
 		format.close();
 	}
 
+	@Test
+	public void testDelimiterOnBufferBoundaryWithWholeFileSplit() throws IOException {
+
+		String[] records = new String[]{"1234567890<DEL?NO!>1234567890", "1234567890<DEL?NO!>1234567890", "<DEL?NO!>"};
+		String delimiter = "<DELIM>";
+		String fileContent = StringUtils.join(records, delimiter);
+
+
+		final FileInputSplit split = createTempFile(fileContent);
+		final FileInputSplit fullSplit = new FileInputSplit(0, split.getPath(), 0, -1, split.getHostnames());
+		final Configuration parameters = new Configuration();
+
+		format.setBufferSize(12);
+		format.setDelimiter(delimiter);
+		format.configure(parameters);
+		format.open(fullSplit);
+
+		for (String record : records) {
+			String value = format.nextRecord(null);
+			assertEquals(record, value);
+		}
+
+		assertNull(format.nextRecord(null));
+		assertTrue(format.reachedEnd());
+
+		format.close();
```

Comment:
nit: this is reductant because it would be handled in `@After`?

Reply:
All test cases include it, so my guess is that the authors also wanted to test close itself and have after only as a fallback. So I'd leave that in.